dashboard/app/controllers/admin_search_controller.rb

```diff
@@ -80,18 +80,23 @@ def show_pilot
     @emails = User.where(id: user_ids).pluck(:email)
   end
 
+  # Parses newline separated emails, ignores commas and whitespace
   def add_to_pilot
-    email = params[:email]
+    emails = params[:email]
     pilot_name = params[:pilot_name]
-    return head :bad_request unless Experiment::PILOT_EXPERIMENTS.find {|p| p[:name] == pilot_name}
-    user = User.find_by_email_or_hashed_email(email)
-    if !user
-      flash[:alert] = "An account with the email address #{email} does not exist"
-    elsif user.student?
-      flash[:alert] = "Cannot add a student to the pilot"
-    else
-      SingleUserExperiment.find_or_create_by!(min_user_id: user.id, name: pilot_name)
-      flash[:notice] = "Successfully added #{email} to #{pilot_name}!"
+    email_array = emails.split("\n").map(&:strip)
+    email_array.each do |email|
+      email = email.gsub(/[\s,]/, "")
+      return head :bad_request unless Experiment::PILOT_EXPERIMENTS.find {|p| p[:name] == pilot_name}
+      user = User.find_by_email_or_hashed_email(email)
+      if !user
+        flash[:alert] = "An account with the email address #{email} does not exist"
+      elsif user.student?
+        flash[:alert] = "Cannot add a student to the pilot"
+      else
+        SingleUserExperiment.find_or_create_by!(min_user_id: user.id, name: pilot_name)
+        flash[:notice] = "Successfully added #{email} to #{pilot_name}!"
```

Comment:
What does this flash notice look like in practice? Are you seeing a notice for each email added? Seems like it might happen too fast for the user to see

Reply:
It appears at the top of the screen after you submit, and isn't all that necessary because the emails show up in the table below anyway. I think it's honestly more useful for the email addresses that get rejected.

Reply:
If you enter a bunch of emails and one of them fails, are you able to actually see that notice before it moves on to the next notice?

Reply:
Yes- the failure stacks below. Maybe it's worth getting rid of the positive/confirmation one entirely?

Reply:
Okay cool - as long as you're able to see the failure, sounds fine to me

Reply:
Sweet. And fwiw- the success alert only displays the most recent email successfully added (not some long list), so I'm going to leave the positive alert too.

Reply:
It would be kind of nice if it displayed all the successes and failures at the end somehow for clarity, but I don't think it's worth the scope increase